_posts/moving-rails-apps-off-webpacker.md

```diff
@@ -0,0 +1,93 @@
+
+[DPUL](https://github.com/pulibrary/dpul/pull/1380) is the only app that DLS has migrated so far. We also tried migrating our largest app [Figgy](https://github.com/pulibrary/figgy), but found ourselves blocked by tricky dependency management in one of the libraries we use. We are planning to remove this library from Figgy eventually, so we set aside the migration for now. Since the blocker came up in the esbuild configuration, it was an issue in both Vite and jsbundling-rails.
+
+The Discovery and Access Services team migrated the Blacklight Catalog application [Orangelight](https://github.com/pulibrary/orangelight/pull/3164) to Vite as well.
+We looked into moving some of our Sprockets usage to Vite as well, but decided to keep our scope limited to the webpacker-to-vite migration.
+With the help of [Vite's migration guide](https://vite-ruby.netlify.app/guide/migration.html#webpacker-%F0%9F%93%A6), this was a smooth transition.
+
+On the Research Data and Scholarship Services (RDSS) team we utilized Vite after DLS and DACS each converted one of thier repositories. [pdc_describe](https://github.com/pulibrary/pdc_describe/pull/397/files) was converted to utilize vite and the transition went smoothly.
+
+#### Vite tips / lessons learned
+
+- Learn to pronounce it!: https://vitejs.dev/guide/#overview
+- Vite does not allow for (or need) testing assets to be served from the [development server](https://github.com/pulibrary/pdc_describe/pull/403).
+- If you import any files that have a file extension other than `.js`, you will need to include
+the file extension when importing it.  We had a few files with an `.es6` file extension, so we
+had to change this:
+
+      import BookCoverManager from '../orangelight/book_covers'
+
+    to this:
+
+      import BookCoverManager from '../orangelight/book_covers.es6'
+
+#### But what's it like to work with?
+
+Migration is only the first step, right? On DLS, we haven't had a work cycle yet
+with the app we moved to vite so we can't speak much to the developer
+experience. It is straightforward to run `bin/vite dev` and have autoreloading
+if you don't want to refresh your browser to see changes.
+
+On DACS, we have made some small changes to Orangelight's javascript since moving to vite,
+and haven't noticed much difference from our previous experience with webpacker.
+
+### Importmaps
+
+Importmaps is supposed to allow you to manage your javascript dependencies
+without having node installed. It's the simplest of the new options so on DLS we
+tried it with our simplest rails app. Like all our other apps, this one uses our
+vue.js-based design system, [lux](https://github.com/pulibrary/lux). We got
+pretty far with importmaps (and dartsass-rails), but we weren't able to get this
+integration with our design system. My speculation is that lux would need to
+provide a node package that bundles all its dependencies. This is partially
```

Comment:
LUX does provide a node package that bundles all its dependencies. That's how it is distributed via npm. Does it need to provide the actual source code of all its dependencies? How would that work with dependencies that aren't pinned?

Reply:
I think so. I don't know exactly :(

Reply:
@sdellis do you think I should change this language in some way?

Reply:
I don't know what a dist file is, really.

Reply:
I updated this.